src/Compilers/Core/Portable/Emit/NoPia/CommonEmbeddedMember.cs

```diff
@@ -69,11 +69,11 @@ private ImmutableArray<TAttributeData> GetAttributes(TPEModuleBuilder moduleBuil
 
                 foreach (var attrData in GetCustomAttributesToEmit(moduleBuilder))
                 {
-                    if (TypeManager.IsTargetAttribute(attrData, AttributeDescription.DispIdAttribute))
+                    if (TypeManager.IsTargetAttribute(attrData, AttributeDescription.DispIdAttribute, out int signatureIndex))
                     {
-                        if (attrData.CommonConstructorArguments.Length == 1)
+                        if (signatureIndex == 0 && TypeManager.TryGetAttributeArguments(attrData, out var constructorArguments, out var namedArguments, syntaxNodeOpt, diagnostics))
```

Comment:
Should we keep checking that `constructorArguments.Length == 1`?

Reply:
> Should we keep checking that ```constructorArguments.Length == 1```?

I intentionally went away from this pattern. The signature index check in combination with the boolean returned by ```TryGetAttributeArguments``` should guarantee that we are dealing with the right constructor and the right set of arguments.